target/riscv/insn32-cheri-64.decode

```diff
@@ -41,16 +41,25 @@ lcddc       1111101  10111 ..... 000 ..... 1011011 @r2  # Would be lqddc for RV1
 lwucap      1111101  01110 ..... 000 ..... 1011011 @r2  # RV64-only
 ldcap       1111101  01011 ..... 000 ..... 1011011 @r2  # RV64-only
 lccap       1111101  11111 ..... 000 ..... 1011011 @r2  # Would be lqcap for RV128
+lr_d_ddc    1111101  10011 ..... 000 ..... 1011011 @atom_ld_cap_or_ddc  # RV64-only (lr_c_ddc for RV32)
+lr_c_ddc    1111101  10100 ..... 000 ..... 1011011 @atom_ld_cap_or_ddc  # RV64-only (would be lr_q_ddc for RV128)
+lr_d_cap    1111101  11011 ..... 000 ..... 1011011 @atom_ld_cap_or_ddc  # RV64-only (lr_c_cap for RV32)
+lr_c_cap    1111101  11100 ..... 000 ..... 1011011 @atom_ld_cap_or_ddc  # RV64-only (would be lr_q_cap for RV128)
+
+sc_d_ddc    1111100  ..... ..... 000 10011 1011011 @atom_st_cap_or_ddc  # RV64-only (sc_c_ddc for RV32)
+sc_d_cap    1111100  ..... ..... 000 11011 1011011 @atom_st_cap_or_ddc  # RV64-only (sc_c_cap for RV32)
+sc_c_ddc    1111100  ..... ..... 000 10100 1011011 @atom_st_cap_or_ddc  # RV64-only (would be sc_q_ddc for RV128)
```

Comment:
The inconsistency between the sorting of loads and stores isn't great, but it's what you've already done for the explicit non-atomic loads and stores in this file... I prefer the sorting used for the loads, as it's also the order the encodings are allocated in (and what you've done for the stores even in `insn32-cheri.decode`).